x/leverage/keeper/interest.go

```diff
@@ -178,6 +192,14 @@ func (k *Keeper) SetLastInterestTime(ctx sdk.Context, interestTime int64) error
 	store := ctx.KVStore(k.storeKey)
 	timeKey := types.CreateLastInterestTimeKey()
 
+	prevTime := k.GetLastInterestTime(ctx)
+
+	if interestTime < prevTime {
```

Comment:
when this is possible?

Reply:
It shouldn't be possible - but there's still that bug mentioned in the comments of AccrueAllInterest, where block time moves backwards. Protecting against it at the `Set` level seems to be a good redundant defense.

`// TODO fix this when tendermint solves https://github.com/tendermint/tendermint/issues/8773`

Reply:
would be good to add a comment to clarify this here. Otherwise it looks strange.